### PR TITLE
Fix Autopet issue

### PR DESCRIPTION
**Pull request by Vladislav94**

### CODE REVIEW

File: src/api/ghst.api.js [hunk 1]
```diff
@@ -24,8 +24,8 @@ export default {
 
     isGhstApproved(address) {
         return contract.allowance(address, AUTOPET_CONTRACT).then(allowance => (
-            ethers.utils.formatUnits(allowance._hex) >= 100
-        ));
+            Number(ethers.utils.formatUnits(allowance._hex)) >= 100
+        ))
```
Comment (dmutrokot) [inline src/api/ghst.api.js:28]: semicolon missed

File: src/pages/Autopet/AutopetContextProvider.js [hunk 2]
```diff
@@ -112,6 +112,15 @@ const AutopetContextProvider = (props) => {
         }
     };
 
+    const checkGhstSpend = async () => {
+        const ghstApproved = await ghstApi.isGhstApproved(connectedWallet);
+
+        if(!ghstApproved) {
```
Comment (dmutrokot) [inline src/pages/Autopet/AutopetContextProvider.js:118]: space missed

File: src/pages/Autopet/AutopetContextProvider.js [hunk 3]
```diff
@@ -125,22 +134,25 @@ const AutopetContextProvider = (props) => {
                 setIsStaked(approval);
                 updateProgress('stake', approval);
                 showSnackbar('success', succesMessage);
+
+                if(!approval) {
```
Comment (dmutrokot) [inline src/pages/Autopet/AutopetContextProvider.js:138]: space missed

File: src/pages/Autopet/AutopetContextProvider.js [hunk 4]
```diff
@@ -125,22 +134,25 @@ const AutopetContextProvider = (props) => {
...
+                    checkGhstSpend()
```
Comment (dmutrokot) [inline src/pages/Autopet/AutopetContextProvider.js:139]: semicolon missed

File: src/pages/Autopet/AutopetContextProvider.js [hunk 5]
```diff
@@ -125,22 +134,25 @@ const AutopetContextProvider = (props) => {
...
+                }
             } else {
-                showSnackbar('error', errorMessage);
-            }
+                showSnackbar('error', errorMessage)
```
Comment (dmutrokot) [inline src/pages/Autopet/AutopetContextProvider.js:142]: semicolon missed

File: src/pages/Autopet/AutopetContextProvider.js [hunk 6]
```diff
@@ -125,22 +134,25 @@ const AutopetContextProvider = (props) => {
...
+            };
 
-            setStakeState('approve');
+            setStakeState('approve')
```
Comment (dmutrokot) [inline src/pages/Autopet/AutopetContextProvider.js:145]: semicolon missed

File: src/pages/Autopet/AutopetContextProvider.js [hunk 7]
```diff
@@ -166,48 +178,33 @@ const AutopetContextProvider = (props) => {
...
+            setIsPetApproved(petApproved);
+            setIsGhstApproved(ghstStaked || ghstApproved);
+            setIsStaked(ghstStaked);
+            setTabs(data => {
+                data.connect.done = walletConnected;
+                data.pet.done = petApproved;
+                data.ghst.done = ghstStaked || ghstApproved;
+                data.stake.done = ghstStaked;
+
+                return {...data}
```
Comment (dmutrokot) [inline src/pages/Autopet/AutopetContextProvider.js:205]: semicolon missed

File: src/pages/Autopet/AutopetContextProvider.js [hunk 8]
```diff
@@ -166,48 +178,33 @@ const AutopetContextProvider = (props) => {
...
+            })
```
Comment (dmutrokot) [inline src/pages/Autopet/AutopetContextProvider.js:206]: semicolon missed

**Review by dmutrokot — CHANGES_REQUESTED** [review]

File: src/pages/Autopet/AutopetContextProvider.js [hunk 9]
```diff
@@ -125,22 +134,25 @@ const AutopetContextProvider = (props) => {
...
                 showSnackbar('success', succesMessage);
+
+                if(!approval) {
+                    checkGhstSpend()
+                }
             } else {
-                showSnackbar('error', errorMessage);
-            }
+                showSnackbar('error', errorMessage)
+            };
```
Comment (dmutrokot) [inline src/pages/Autopet/AutopetContextProvider.js:143]: no semicolon needed

**Review by dudendy — APPROVED** [review]

**Pull request merged**